client/rest-high-level/build.gradle

```diff
@@ -59,4 +59,17 @@ forbiddenApisMain {
   // core does not depend on the httpclient for compile so we add the signatures here. We don't add them for test as they are already
   // specified
   signaturesURLs += [PrecommitTasks.getResource('/forbidden/http-signatures.txt')]
+}
+
+javadoc {
+    dependsOn ':core:javadoc'
+    dependsOn ':client:rest:javadoc'
+    String host;
+    if (version.endsWith("-SNAPSHOT")) {
+        host = "https://snapshots.elastic.co";
+    } else {
+        host = "https://artifacts.elastic.co";
+    }
```

Comment:
suggestions are welcome on where this repeated if can be moved and shared across the three projects that need it.